src/cpp/core/r_util/RActiveSessionsStorage.cpp

```diff
@@ -0,0 +1,149 @@
+/*
+ * RActiveSessionsStorage.cpp
+ *
+ * Copyright (C) 2022 by RStudio, PBC
+ *
+ * Unless you have received this program directly from RStudio pursuant
+ * to the terms of a commercial license agreement with RStudio, then
+ * this program is licensed to you under the terms of version 3 of the
+ * GNU Affero General Public License. This program is distributed WITHOUT
+ * ANY EXPRESS OR IMPLIED WARRANTY, INCLUDING THOSE OF NON-INFRINGEMENT,
+ * MERCHANTABILITY OR FITNESS FOR A PARTICULAR PURPOSE. Please refer to the
+ * AGPL (http://www.gnu.org/licenses/agpl-3.0.txt) for more details.
+ *
+ */
+
+#include <core/r_util/RActiveSessionsStorage.hpp>
+#include <core/r_util/RActiveSessionStorage.hpp>
+#include <core/r_util/RActiveSessions.hpp>
+#include <algorithm>
+
+#define kSessionDirPrefix "session-"
+
+namespace rstudio {
+namespace core {
+namespace r_util {
+
+namespace
+{
```

Comment:
Prefer constexpr for literals to macros.

```suggestion
namespace rstudio {
namespace core {
namespace r_util {

namespace
{

constexpr const char * kSessionDirPrefix = "session-";

```